pkg/api/plugin.go

```diff
@@ -119,4 +119,8 @@ type Plugin interface {
 
 	// GetControlPlanePods fetches a consolidated list of the control plane pods in the cluster
 	GetControlPlanePods(ctx context.Context, oc *OpenShiftManagedCluster) ([]byte, error)
+
+	// ReimageAndRestart reimages and restarts a virtual machine within a scale set in the cluster.
+	// wipeTempDisks indicates whether temporary disks on the VM should be reimaged/wiped as well.
+	ReimageAndRestart(ctx context.Context, oc *OpenShiftManagedCluster, vmScaleSetName string, instanceId string, wipeTempDisks bool) *PluginError
```

Comment:
When preparing request you are using `OpenShiftManagedClustersVMReimageAndRestartParameters` and encode them into the payload. But the method does not accept anything like that. So you set parameters, encode them into the payload, but never use them in the fakeRP. I get why, but I think we should be aligned with code we produce and how we use it. And this cries for tests....

Reply:
remove the wipeTempDisk argument?

Reply:
Or use the parameters object and parse it inside and decide.

Reply:
the reason for the `wipeTempDisk` parameter here is so I can do this: https://github.com/openshift/openshift-azure/pull/1091/commits/10aa7182735ebd0ae5586ff8b82c874b3b13c2db#diff-643883fea487eafcb41824e842b9fd36R10